### PR TITLE
feat: 🎸 [HCPSDKFIORIUIKIT-2791]SwiftUI: floating value support

### PR DESCRIPTION
**Pull request by zzchao-1999**
Screenshot 2024-10-18 at 17 14 37

### CODE REVIEW

File: Sources/FioriSwiftUICore/_ComponentProtocols/CompositeComponentProtocols.swift [hunk 1]
```diff
@@ -133,10 +133,14 @@ protocol _JouleWelcomeScreen: _MediaImageComponent, _GreetingTextComponent, _Tit
 protocol _StepperFieldComponent: _DecrementActionComponent, _TextInputFieldComponent, _IncrementActionComponent {
     /// The step value
     // sourcery: defaultValue = 1
-    var step: Int { get }
+    var step: Double { get }
     
     /// a range of values
-    var stepRange: ClosedRange<Int> { get }
+    var stepRange: ClosedRange<Double> { get }
```
Comment (billzhou0223) [inline Sources/FioriSwiftUICore/_ComponentProtocols/CompositeComponentProtocols.swift:139]: How do I control the interval of the stepper?

**Review by billzhou0223 — CHANGES_REQUESTED** [review]

File: Sources/FioriSwiftUICore/_ComponentProtocols/CompositeComponentProtocols.swift (hunk 1 above)
Comment (zzchao-1999) [reply Sources/FioriSwiftUICore/_ComponentProtocols/CompositeComponentProtocols.swift:139]: Did you mean the increment/decrement size? If so, you can use the "step" parameter.
var step: Double { get }

**Comment by zzchao-1999** [comment]
> We may need to enhance the display of number of decimals in the StepperField. e.g. if you have a range of 0 to 1.0, and stepper value is 0.1, when you increase/decrease, you may see values as "0.6999999999999". This is due to "Double" is used as the new data type. One suggestion could be, we can check the stepper value. Then use the same number of decimals, to format the StepperField value

Added a number formatter to ensure the display of number of decimals in the stepperField.

**Review by dyongxu — APPROVED** [review]

**Review by billzhou0223 — APPROVED** [review]

**Comment by dyongxu** [comment]
@zzchao-1999 Check with the reporter if it is needed for rel-4.1

**Pull request merged**